sqlmesh/core/engine_adapter/base.py

```diff
@@ -366,22 +398,20 @@ def create_view(
             create_kwargs: Additional kwargs to pass into the Create expression
         """
         schema: t.Optional[exp.Table | exp.Schema] = exp.to_table(view_name)
-
-        if isinstance(query_or_df, DF_TYPES):
-            if PySparkDataFrame is not None and isinstance(query_or_df, PySparkDataFrame):
-                query_or_df = query_or_df.toPandas()
-
-            if not isinstance(query_or_df, pd.DataFrame):
-                raise SQLMeshError("Can only create views with pandas dataframes.")
-
+        df = self.try_get_pandas_df(query_or_df)
+        pyspark_df = self.try_get_pyspark_df(query_or_df)
```

Comment:
I wonder whether this logic should even be in the base class. Can this be moved into the spark version of the class?

Reply:
Yeah I felt mixed about this too. Technically we can do this since we duck type now (don't have to worry about import errors) so this is fairly simple approach but yeah it does mean spark stuff is making it's way into the base. Will change.